### PR TITLE
fix: Raise Invalid Resource When DisableExecuteApiEndpoint: False And Has No DefinitionBody

### PR DESCRIPTION
**Pull request by CoshUS**
*Issue #, if available:*
When DisableExecuteApiEndpoint is False, http api and api generator will treat it as None which means the default, True.

*Description of changes:*

*Description of how you validated changes:*

*Checklist:*

- [ ] Add/update tests using:
    - [ ] Correct values
    - [ ] Bad/wrong values (None, empty, wrong type, length, etc.)
    - [ ] [Intrinsic Functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html)
- [ ] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: bin/sam-translate.py [hunk 1]
```diff
@@ -28,6 +26,9 @@
 import boto3
 from docopt import docopt
 
+if sys.version_info.major == 3:
```
Comment (hawflau) [inline bin/sam-translate.py:29]: We can drop this version check and just import the module as we are already py3 only on the `develop` branch

File: bin/sam-translate.py [hunk 2]
```diff
@@ -1,5 +1,3 @@
-#!/usr/bin/env python2
```
Comment (hawflau) [inline bin/sam-translate.py:1]: We still want to keep this and update from `python2` to `python` as this file is executable.

**Review by hawflau — CHANGES_REQUESTED** [review]

File: bin/sam-translate.py (hunk 1 above)
Comment (CoshUS) [reply bin/sam-translate.py]: Removed.

File: bin/sam-translate.py (hunk 2 above)
Comment (CoshUS) [reply bin/sam-translate.py:1]: Added back as `python`.

**Review by hoffa — APPROVED** [review]

**Review by hawflau — APPROVED** [review]

**Pull request merged**